src/HotChocolate/Fusion-vnext/src/Fusion.Composition/Results/Result.cs

```diff
@@ -0,0 +1,52 @@
+namespace HotChocolate.Fusion.Results;
+
+public readonly record struct Result
```

Comment:
This name might be too generic.  I have done the same mistake in GreenDonut.

Reply:
It was `CompositionResult` at one stage, but now it's not specific to composition, and could actually be shared.

I can rename it if you'd prefer not to create a shared lib at this stage, but there will be duplication.